specs/lightclient/beacon-chain.md

```diff
@@ -0,0 +1,208 @@
+    compact_validators: List[uint64, MAX_VALIDATORS_PER_COMMITTEE]
+```
+
+## Helper functions
+
+### Misc
+
+
+#### `pack_compact_validator`
+
+```python
+def pack_compact_validator(index: ValidatorIndex, slashed: bool, balance_in_increments: uint64) -> uint64:
+    """
+    Create a compact validator object representing index, slashed status, and compressed balance.
+    Takes as input balance-in-increments (// EFFECTIVE_BALANCE_INCREMENT) to preserve symmetry with
+    the unpacking function.
+    """
+    return (index << 16) + (slashed << 15) + balance_in_increments
+```
+
+#### `unpack_compact_validator`
+
+```python
+def unpack_compact_validator(compact_validator: uint64) -> Tuple[ValidatorIndex, bool, uint64]:
+    """
+    Return validator index, slashed, balance // EFFECTIVE_BALANCE_INCREMENT
+    """
+    return (
+        ValidatorIndex(compact_validator >> 16),
+        bool((compact_validator >> 15) % 2),
+        compact_validator & (2**15 - 1),
+    )
+```
+
+#### `committee_to_compact_committee`
+
+```python
+def committee_to_compact_committee(state: BeaconState, committee: Sequence[ValidatorIndex]) -> CompactCommittee:
+    """
+    Given a state and a list of validator indices, outputs the ``CompactCommittee`` representing them.
+    """
+    validators = [state.validators[i] for i in committee]
+    compact_validators = [
+        pack_compact_validator(i, v.slashed, v.effective_balance // EFFECTIVE_BALANCE_INCREMENT)
+        for i, v in zip(committee, validators)
+    ]
+    pubkeys = [v.pubkey for v in validators]
+    return CompactCommittee(pubkeys=pubkeys, compact_validators=compact_validators)
+```
+
+### Beacon state accessors
+
+#### `get_light_client_committee`
+
+```python
+def get_light_client_committee(beacon_state: BeaconState, epoch: Epoch) -> Sequence[ValidatorIndex]:
+    """
+    Return the light client committee of no more than ``LIGHT_CLIENT_COMMITTEE_SIZE`` validators.
+    """
+    source_epoch = (max(epoch // LIGHT_CLIENT_COMMITTEE_PERIOD, 1) - 1) * LIGHT_CLIENT_COMMITTEE_PERIOD
+    active_validator_indices = get_active_validator_indices(beacon_state, source_epoch)
+    seed = get_seed(beacon_state, source_epoch, DOMAIN_LIGHT_CLIENT)
+    return [
+        compute_shuffled_index(i, active_validator_indices, seed)
+        for i in range(min(active_validator_indices, LIGHT_CLIENT_COMMITTEE_SIZE))
+    ]
+```
+
+### Block processing
+
+```python
+def process_block(state: BeaconState, block: BeaconBlock) -> None:
+    phase0.process_block(state, block)
+    process_light_client_signature(state, block.body)
+```
+
+#### Light client processing
+
+```python
+def process_light_client_signature(state: BeaconState, block_body: BeaconBlockBody) -> None:
+    committee = get_light_client_committee(state, get_current_epoch(state))
+    previous_slot = max(state.slot, 1) - 1
+    previous_block_root = get_block_root_at_slot(state, previous_slot)
+
+    # Light clients sign over the previous block root
```

Comment:
May be a heavy-handed edit but I'm trying to sort out what kind of security guarantees we get here (and how) so I don't think it hurts to be as precise as possible...

```suggestion
    # Light client committees sign over the previous block root
```

Reply:
,my name is Aura ,I saved your group page long time ago ,after years hope all is good with you .
Always believed in it so I had 0.5 Ethereum that I keep it for a long time for my daughter and lost it to a scammer. 
Now after I m putting the baby to sleep I m reading and watching videos. 
Is any chance to recover it?.
I have 200$ I know is not much and I know is late to invest ?
I know but I dont have much.

Thank youuu